server/src/main/java/org/elasticsearch/repositories/RepositoriesService.java

```diff
@@ -142,16 +142,7 @@ public RepositoriesService(
     public void registerRepository(final PutRepositoryRequest request, final ActionListener<AcknowledgedResponse> listener) {
         assert lifecycle.started() : "Trying to register new repository but service is in state [" + lifecycle.state() + "]";
 
-        final RepositoryMetadata newRepositoryMetadata = new RepositoryMetadata(request.name(), request.type(), request.settings());
-        validate(request.name());
-
-        // Trying to create the new repository on master to make sure it works
-        try {
-            closeRepository(createRepository(newRepositoryMetadata, typesRegistry, RepositoriesService::throwRepositoryTypeDoesNotExists));
-        } catch (Exception e) {
-            listener.onFailure(e);
-            return;
-        }
+        validateRepository(request, listener);
```

Comment:
The validation logic was extracted so we can use it in the reserved state handler.

Reply:
This seems broken, we should return here on any validation exception?
Maybe better to leave this a non-async method and do the try catch here and let the exception bubble up in the new spot this is used in where the listener is somewhat artificial anyway?

Reply:
So while fixing this I noticed something, the name validation is a sync failure, but creating the repository uses the listener. I'm not sure there's a reason for this difference, but I changed the code to mimic the original behaviour now.